src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *               type: array
+ *               items:
+ *                 type: object
+ *                 properties:
+ *                   _id:
+ *                     type: string
+ *                     description: The unique identifier of the post.
+ *                   title:
+ *                     type: string
+ *                     description: The title of the post.
+ *                   content:
+ *                     type: string
+ *                     description: The content of the post.
+ *                   createdAt:
+ *                     type: string
+ *                     format: date-time
+ *                     description: The creation timestamp of the post.
+ *                   creator_id:
+ *                     type: object
+ *                     properties:
+ *                       _id:
+ *                         type: string
+ *                         description: The unique identifier of the creator.
+ *                       username:
+ *                         type: string
+ *                         description: The username of the post creator.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message.
+ */
 router.get('/userPosts/:userID', getUserPosts);
+
+/**
+ * @swagger
+ * /posts/count/categories:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Count posts by category
+ *     description: Retrieves the count of posts grouped by their categories. This endpoint aggregates posts to provide a count for each category.
+ *     responses:
+ *       200:
+ *         description: A list of categories with the count of posts in each category.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
+ *               items:
+ *                 type: object
+ *                 properties:
+ *                   name:
+ *                     type: string
+ *                     description: The name of the category.
+ *                   count:
+ *                     type: integer
+ *                     description: The number of posts in the category.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message.
+ *                 err:
+ *                   type: string
+ *                   description: Details of the error.
+ */
 router.get('/count-by-category', countPostsByCategory);
 
+/**
+ * @swagger
+ * /posts:
+ *   post:
+ *     tags:
+ *       - Posts operations
+ *     summary: Upload a new post , Must be connected as user
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"